Jenkinsfile

```diff
@@ -20,18 +18,34 @@ pipeline {
             steps {
                 sh '''
                     mkdir -p /TheSaiyan/Contact-Manager/
+                    JAR_NAME=contactManager.jar
                     cp build/libs/contactmanager-0.0.1-SNAPSHOT.jar ${JAR_NAME}
                 '''
             }
         }
 
         stage('Start the service') {
             steps {
+
                 sh '''
-                    sudo supervisorctl stop contact_manager
-                    gzip /TheSaiyan/Contact-Manager/${APP_NAME}*.jar
-                    mv ${JAR_NAME}.jar /TheSaiyan/Contact-Manager/
-                    sudo supervisorctl restart contact_manager
+                    supervisorctl stop contact_manager
+                    sleep 1s
+
+                    JAR_NAME=contactManager.jar
+
+                    ARCH_JAR_NAME=contactManager-`date +'%Y%m%d'`.jar
+
+                    if [ -f /TheSaiyan/Contact-Manager/${JAR_NAME} ]; then
+                        mv /TheSaiyan/Contact-Manager/${JAR_NAME} /TheSaiyan/Contact-Manager/${ARCH_JAR_NAME}
+                        gzip -f /TheSaiyan/Contact-Manager/${ARCH_JAR_NAME}
+                    fi
+                    sleep 2s
+
+                    if [ -f ${JAR_NAME} ]; then
+                        mv ${JAR_NAME} /TheSaiyan/Contact-Manager/
+                    fi
+
+                    supervisorctl start contact_manager
```

Comment:
1. Stops the current running process
2. Give a delay of 1 second
3. Rename the current running Jar and suffixed it with current date
4. Zip the renamed jar
5. Move the new jar from the `workspce` to `/TheSaiyan/Contact-Manager/` folder
6. Start the service again